src/borg/archive.py

```diff
@@ -1557,7 +1557,7 @@ def is_temporary_archive(archive_name):
 
     def __init__(self, repository, manifest, key, cache, matcher,
                  exclude_caches=False, exclude_if_present=None, keep_exclude_tags=False,
-                 chunker_params=None, compression=None, always_recompress=False,
+                 chunker_params=None, compression=None, recompress=False, always_recompress=False,
```

Comment:
Yeah that seems better than having two independent flags. Perhaps argparse can make it so that just `--recompress` means if-different to abbreviate the common case. (Well, as "common" as recreate can be).